console/src/views/Radio.js

```diff
@@ -0,0 +1,193 @@
+import Button from '@enact/agate/Button';
+import Divider from '@enact/agate/Divider';
+import {LabeledItemBase} from '@enact/agate/LabeledItem';
+import {Panel} from '@enact/agate/Panels';
+import ToggleButton from '@enact/agate/ToggleButton';
+import kind from '@enact/core/kind';
+import React, {Component} from 'react';
+
+import PresetItem from '../components/PresetItem';
+
+import css from './Radio.less';
+
+
+const RadioBase = kind({
+	name: 'Radio',
+
+	handlers: {
+		onFrequencyToggle: (ev, {changeFrequency}) => {
+			const frequency = ev.currentTarget.innerText.trim();
+			changeFrequency(frequency);
+		},
+		onPresetClick: (ev, {changeStation, presets}) => {
+			changeStation(presets[ev.presetIndex]);
+		},
+		onPresetDown: (ev, {changePreset}) => {
+			changePreset(ev.presetIndex);
+		},
+		onPresetHold: (ev, {currentStation, currentPreset, updatePresets}) => {
+			updatePresets(currentStation, currentPreset);
+		},
+		onTune: (ev, {currentStation, changeStation}) => {
+			const action = ev.currentTarget.getAttribute('action');
+			let newStation;
+
+			if (action === 'tune-up') {
+				if (currentStation >= 108) {
+					newStation = 87.8;
+				} else {
+					newStation = (currentStation*10 + 1)/10;
+				}
+			} else if (action === 'tune-down') {
+				if (currentStation <= 87.8) {
+					newStation = 108;
+				} else {
+					newStation = (currentStation*10 - 1)/10;
+				}
+			} else if (action === 'scan-up') {
+				if (currentStation >= 108) {
+					newStation = 87.8;
+				} else {
+					newStation = (currentStation*10 + 20)/10;
+				}
+			} else if (action === 'scan-down') {
+				if (currentStation <= 87.8) {
+					newStation = 108;
+				} else {
+					newStation = (currentStation*10 - 20)/10
+				}
+			}
+			changeStation(newStation);
+		}
+	},
+
+	render: ({currentStation, frequency, presets, onFrequencyToggle, onPresetClick, onPresetDown, onPresetHold, onTune}) => (
+		<Panel>
+			<div className={css.radio}>
+				<div className={css.radioToggle}>
+					<ToggleButton onClick={onFrequencyToggle} selected={frequency === 'AM'} small type="grid">AM</ToggleButton> |
+					<ToggleButton onClick={onFrequencyToggle} selected={frequency === 'FM'} small type="grid">FM</ToggleButton>
+				</div>
+				<div className={css.title}>
+					{/*Radio TextInfo*/}
+					<LabeledItemBase label="Artist - Song">{currentStation} MHZ</LabeledItemBase>
+				</div>
+				<div className={css.tuneControls}>
+					{/*Tune*/}
+					<Button onClick={onTune} action="tune-down" icon={"arrowsmallleft"} />
+					Tune
+					<Button onClick={onTune} action="tune-up" icon={"arrowsmallright"} />
+				</div>
+				<div className={css.scanControls}>
+					<Button onClick={onTune} action="scan-down" icon={"arrowsmallleft"} />
+					Scan
+					<Button onClick={onTune} action="scan-up" icon={"arrowsmallright"} />
+				</div>
+				{/*List*/}
+				<div className={css.presetList}>
+					Presets
+					<Divider startSection />
+					<PresetItem
```

Comment:
`ui/Repeater` would be handy here to reduce duplication. Might require some data reordering to match up to `PresetItem`'s API.